fraudproofs/badencodingfraudproof.go

```diff
@@ -0,0 +1,96 @@
+package fraudproofs
+
+import (
+	"errors"
+	"github.com/celestiaorg/celestia-core/state/txindex/null"
+	tmproto "github.com/proto/tendermint/types"
+	nmt "github.com/celestiaorg/nmt"
+	tmhash "github.com/celestiaorg/lazyledger-core/crypto/tmhash"
+	"github.com/celestiaorg/rsmt2d"
+)
+
+
+func VerifyBadEncodingFraudProof(proof tmproto.BadEncodingFraudProof) (bool, error) {
+
+	// get the block
+	//TODO
+	
+	// parse the bad encoding fraud proof
+	isColForProof := proof.GetIsCol()
+	height := proof.GetHeight()
+	shareProofs := proof.GetShareProofs()
+	position := proof.GetPosition()
+
+	// check if shareProofs has correct size
+	if len(shareProofs) != AVAILABLE_DATA_ORIGINAL_SQUARE_MAX {
+		return errors.New("Number of shares provided is incorrect.")
+	}
+
+	// get the dataRoot within the header
+	dataRoot := block.GetHeader().GetDataHash()
+
+	// get the row or column root challenged by the fraud proof within the DA header
+	dataAvailabilityHeader := block.GetDataAvailabilityHeader()
+	axisRoot := nil
+	if isColForProof {
+		columnRoots := dataAvailabilityHeader.GetColumnRoots()
+		if (position < len(columnRoots)) && (0 <= position) {
+			axisRoot = columnRoots[position] 
+		}
+		else {
+			return errors.New("Position out of bounds in the badencodingfraudproof.")
+		}
+	}
+	else {
+		rowRoots := dataAvailabilityHeader.GetRowRoots()
+		if (position < len(rowRoots)) && (0 <= position) {
+			axisRoot = rowRoots[position] 
+		}
+		else {
+			return errors.New("Position out of bounds in the badencodingfraudproof.")
+		}
+	}
+
+	// new namespacedMerkleTree for calculating the new root
+	namespacedMerkleTree := nmt.New(tmhash.New())
+	err := nil
```

Comment:
There is no need to define this var here. Define it where it is used (e.g. `err := namespacedMerkleTree.push(share)` instead of `err = namespacedMerkleTree.push(share)` below).